bpf/bpf_lxc.c

```diff
@@ -635,8 +635,15 @@ static inline int handle_ipv4_from_lxc(struct __sk_buff *skb, __u32 *dstID)
 
 #ifdef ENCAP_IFINDEX
 	if (tunnel_endpoint) {
-		return encap_and_redirect_with_nodeid(skb, tunnel_endpoint,
-						      SECLABEL, monitor);
+		int ret = encap_and_redirect_with_nodeid_from_lxc(skb, tunnel_endpoint,
+								  SECLABEL, monitor);
+
+		if (ret == TC_ACT_REDIRECT)
+			return ret;
+		/* If not redirected noteably due to IPSEC then pass up to stack
+		 * for further processing.
+		 */
+		goto pass_to_stack;
```

Comment:
hmm OK we should catch DROP_WRITE error case as well. Then I think we do not need to do anything to differentiate between ipsec and non-ipsec. So I think the error here is not properly handling the error cases and dropping pkt.

Reply:
```
                /* Redirected by encap code path */
                if (ret == TC_ACT_REDIRECT)
                        return ret;
                /* If not redirected noteably due to IPSEC then pass up to stack
                 * for further processing.
                 */
                else if (!ret)
                        goto pass_to_stack;
                /* Otherwise this is an error and packet is being dropped. */
                else
```